connect/api/src/main/java/org/apache/kafka/connect/source/SourceRecord.java

```diff
@@ -46,25 +46,49 @@
     private final Map<String, ?> sourcePartition;
     private final Map<String, ?> sourceOffset;
 
+    /**
+     * @deprecated use {@link SourceRecord.Builder}
+     */
+    @Deprecated
```

Comment:
@hachikuji depends... the larger the number of optional parameters, the more the value in Builder as a way to have named & default arguments in Java